packages/adapter/src/response-processing/encoding.ts

```diff
@@ -1,42 +1,54 @@
 import { ethers } from 'ethers';
 import { ResponseType, ValueType } from '../types';
 
-export function convertUnsignedIntegerToBytes32(value: string) {
+export function convertUnsignedInteger(value: string) {
   return ethers.utils.defaultAbiCoder.encode(['uint256'], [value]);
 }
 
-export function convertSignedIntegerToBytes32(value: string) {
+export function convertSignedInteger(value: string) {
   return ethers.utils.defaultAbiCoder.encode(['int256'], [value]);
 }
 
-function trimValue(value: string) {
+export function convertBool(value: boolean) {
+  return ethers.utils.defaultAbiCoder.encode(['bool'], [value]);
+}
+
+export function convertBytes32(value: string) {
   // We can't encode strings longer than 31 characters to bytes32.
   // Ethers need to keep room for null termination
-  return value.length > 31 ? value.substring(0, 31) : value;
-}
+  const trimmedValue = value.length > 31 ? value.substring(0, 31) : value;
 
-export function convertStringToBytes32(value: string) {
-  const trimmedValue = trimValue(value);
   const bytes32String = ethers.utils.formatBytes32String(trimmedValue);
   return ethers.utils.defaultAbiCoder.encode(['bytes32'], [bytes32String]);
 }
 
-export function convertBoolToBytes32(value: boolean) {
-  return ethers.utils.defaultAbiCoder.encode(['bool'], [value]);
+export function convertAddress(value: string) {
+  return ethers.utils.defaultAbiCoder.encode(['address'], [value]);
```

Comment:
Should we check if `value` is actually a valid `address`? https://docs.ethers.io/v5/api/utils/address/#utils-isAddress

Or maybe this should be done in a specific `castAddress()` function in [casting.ts](https://github.com/api3dao/airnode/blob/b83870b816f7ad7c36747844b80d23387976c6ec/packages/adapter/src/response-processing/casting.ts) instead of using `castStringLike()`.

Reply:
Yeah, that is probably a good idea.